src/platform-includes/distributed-tracing/custom-instrumentation/java.mdx

```diff
@@ -0,0 +1,66 @@
+On this page you will learn how to manually propagate trace information into and out of your Java application. Please note, that you do not need to do this manually, if you [use one of our supported frameworks](/platforms/java/usage/distributed-tracing/#how-to-use-distributed-tracing), or you have our <PlatformLink to="/performance/">performance monitoring feature</PlatformLink> turned on.
+
+To set it up manually, all you have to do is to make sure your application extracts incoming headers and to set those headers again when making an outgoing request within your application.
+
+## Step 1) Extract Incoming Tracing Information
+
+Incoming tracing information has to be extracted and stored in memory for later use. Sentry provides the `continueTrace()` function to help you with this. Incoming tracing information can come from different places:
+
+- In a web environment it will be sent with HTTP headers, for example, by another Sentry SDK used in your frontend project.
+
+Here's an example of how to extract and store incoming tracing information using `continueTrace()`:
+
+```java
+import io.sentry.BaggageHeader;
+import io.sentry.ITransaction;
+import io.sentry.Sentry;
+import io.sentry.SentryTraceHeader;
+import io.sentry.TransactionContext;
+
+final String sentryTraceHeader = httpRequest.getHeader(SentryTraceHeader.SENTRY_TRACE_HEADER);
+final List<String> baggageHeader = Collections.list(httpRequest.getHeaders(BaggageHeader.BAGGAGE_HEADER));
+
+final TransactionContext transactionContext = Sentry.continueTrace(sentryTraceHeader, baggageHeader);
+if (transactionContext != null) {
+    final ITransaction transaction = Sentry.startTransaction(transactionContext);
+}
+```
+
+In this example, `getHeader()` returns a single `String` or `null` and `getHeaders()` returns an enumeration of `Strings` which may be empty.
+
+By passing these headers to Sentry's `continueTrace()` function it will store them in memory for later use.
+
+## Step 2) Inject Tracing Information to Outgoing Requests
+
+For distributed tracing to work, the two headers `sentry-trace` and `baggage`, must now also be added to outgoing requests.
+
+### Inject Tracing Information Into HTTP Requests
+
+If you are sending outgoing HTTP requests with <PlatformLink to="/guides/spring/performance/instrumentation/automatic-instrumentation/">RestTemplate</PlatformLink>, [WebClient](https://docs.spring.io/spring-framework/reference/web/webflux-webclient.html), <PlatformLink to="/configuration/integrations/okhttp/">OkHttp</PlatformLink>, <PlatformLink to="/performance/instrumentation/apollo/">Apollo</PlatformLink>, <PlatformLink to="/performance/instrumentation/apollo3/">Apollo 3</PlatformLink>, <PlatformLink to="/performance/instrumentation/open-feign/">OpenFeign</PlatformLink> and have our integration for it enabled, this tracing information is automatically added to outgoing requests.
```

Comment:
Please revisit the links, some of them return a 404. Also, we should not point the users to the performance docs for the custom instrumentation, since it would mean they have to activate performance monitoring, right? And if they have activated it anyhow, they do not need the custom instrumentation docs. Or am I missing something?

Reply:
Build step was also failing for links: https://github.com/getsentry/sentry-docs/actions/runs/5445750999/jobs/9905449599?pr=7313

Reply:
> Please revisit the links, some of them return a 404
> Build step was also failing for links

Yeah some links are for Android. That's probably why they're not working here since this is Java. Just wanted to wait for a review before starting to copy all the stuff.

> Also, we should not point the users to the performance docs for the custom instrumentation, since it would mean they have to activate performance monitoring, right?

I only wanted to point them toward the right dependency to add to their project but yeah this would be confusing. Is it OK to just add another sentence like:

Reply:
```suggestion
If you are sending outgoing HTTP requests with <PlatformLink to="/guides/spring/performance/instrumentation/automatic-instrumentation/">RestTemplate</PlatformLink>, [WebClient](https://docs.spring.io/spring-framework/reference/web/webflux-webclient.html), <PlatformLink to="/configuration/integrations/okhttp/">OkHttp</PlatformLink>, <PlatformLink to="/performance/instrumentation/apollo/">Apollo</PlatformLink>, <PlatformLink to="/performance/instrumentation/apollo3/">Apollo 3</PlatformLink>, <PlatformLink to="/performance/instrumentation/open-feign/">OpenFeign</PlatformLink> and have our integration for it enabled, this tracing information is automatically added to outgoing requests. NOTE: You do not have to enable Performance for distributed tracing to work.
```

Reply:
I removed the links for some of them for now. For Android we have https://docs.sentry.io/platforms/android/configuration/integrations/ but for Java that's pretty much empty. Maybe instead of linking performance subpages (which don't work for all guides anyways) we should add integrations to that page for Java at some point.